libsnark/zk_proof_systems/plonk/srs.hpp

```diff
@@ -134,13 +131,14 @@ template<typename ppT> class srs
         const libff::Fr<ppT> &k2,
         std::vector<libff::G1<ppT>> &&secret_powers_g1,
         std::vector<libff::G2<ppT>> &&secret_powers_g2,
-        const polynomial<Field> &L_basis_zero,
-        std::shared_ptr<libfqfft::evaluation_domain<Field>> domain);
+        const polynomial<Field> &L_basis_zero);
 };
 
 template<typename ppT>
 srs<ppT> plonk_srs_derive_from_usrs(
-    const usrs<ppT> &usrs, const circuit_t<ppT> &circuit);
+    const usrs<ppT> &usrs,
+    const circuit_t<ppT> &circuit,
+    std::shared_ptr<libfqfft::evaluation_domain<libff::Fr<ppT>>> domain);
```

Comment:
Can the domain be computed inside this function?  Otherwise the user of the plonk code must create it.  Domain should be deterministic, given the usrs / srs, so the user of the plonk code should not have to do it.

Similar comments hold for prover and verifier.  I don't think it needs to appear in the signatures of the top-level functions exposed by this code module.  But let me know if I've missed something here.

Reply:
Yes.  That seems right to me.

Reply:
Done and pushed.